core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceSpendReportComponent.scala

```diff
@@ -0,0 +1,304 @@
+      new DateTime(zonedDateTime.toInstant.toEpochMilli)
+    }
+
+  def fromSpendReportingResults(spendReportingResults: SpendReportingResults): Seq[WorkspaceSpendReport] = {
+    val summary = spendReportingResults.spendSummary
+    spendReportingResults.spendDetails.flatMap { spendDetail =>
+      spendDetail.spendData.map {
+        var totalStorage: Option[Float] = None
+        var totalCompute: Option[Float] = None
+        var otherSpend: Option[Float] = None
+        var storageCredits: Option[Float] = None
+        var computeCredits: Option[Float] = None
+        var otherCredits: Option[Float] = None
+        spendData =>
+          spendData.subAggregation.get.spendData.foreach { categorySpend =>
+            val categoryCost = Some(categorySpend.cost.toFloat)
+            val categoryCredits = Some(categorySpend.credits.toFloat)
+            categorySpend.category match {
+              case Some(TerraSpendCategories.Storage) =>
+                totalStorage = categoryCost
+                storageCredits = categoryCredits
+              case Some(TerraSpendCategories.Compute) =>
+                totalCompute = categoryCost
+                computeCredits = categoryCredits
+              case Some(TerraSpendCategories.Other) =>
+                otherSpend = categoryCost
+                otherCredits = categoryCredits
+              // This is not included in the consolidated spend report
+              case Some(TerraSpendCategories.WorkspaceInfrastructure) =>
+              case None                                               =>
+            }
+          }
+          WorkspaceSpendReport.newWorkspaceSpendReport(
+            spendData.googleProjectId.get.value,
+            convertJodaToJava(summary.startTime),
+            convertJodaToJava(summary.endTime),
+            summary.currency,
+            isDataAvailable = true,
+            totalCompute,
+            totalStorage,
+            otherSpend,
+            computeCredits,
+            storageCredits,
+            otherCredits
+          )
+      }
+    }
+  }
+
+  def toSpendReportingResults(records: Seq[WorkspaceSpendReport],
+                              projectNames: Map[GoogleProjectId, WorkspaceName]
+  ): SpendReportingResults = {
+    var start: Option[DateTime] = None
+    var end: Option[DateTime] = None
+    var total_spend = BigDecimal(0.0)
+    var total_credits = BigDecimal(0.0)
+    val currency = records.map(_.currency).distinct match {
+      case head :: List() => Currency.getInstance(head)
+      case head :: tail =>
+        throw RawlsExceptionWithErrorReport(
+          StatusCodes.InternalServerError,
+          s"Inconsistent currencies found while aggregating spend data: $head and ${tail.head} cannot be combined"
+        )
+      case List() => throw RawlsExceptionWithErrorReport(StatusCodes.NotFound, "No currencies found for spend data")
+    }
+
+    val all = records.map { record =>
```

Comment:
This chunk of code seems pretty similar to what's in `SpendReportingService`.  Is there a way it can be reused?